pypdf/_xobj_image_helpers.py

```diff
@@ -125,6 +125,20 @@ def _get_imagemode(
     return mode, mode == "CMYK"
 
 
+def _extended_image_frombytes(
+    mode: str, size: Tuple[int, int], data: bytes
+) -> Any:  # Any = ImageType
+    try:
+        img = Image.frombytes(mode, size, data)
+    except ValueError as exc:
+        if len(data) % (size[0] * size[1]) != 0:
+            raise exc
+        k = size[0] * size[1] * len(mode) / len(data)
+        data = b"".join([bytes((x,) * int(k)) for x in data])
```

Comment:
Could you please explain what this does and why this works? As far as I understand it, this will create duplicates of each byte, but I do not really understand why this fixes our issues and why we can assume that rounding to the floor is correct here.

Reply:
the issue comes from the fact that the data contains only 1 pixel with 8 bits data and not 8bits for each color/band. It is the original issue reported  in #2343. I've found no reasons in the fields to establish that data should be duplicated. I've preferred the solution to use the exception handling to prevent affecting the nominal case